corehq/apps/cloudcare/static/cloudcare/js/formplayer/utils/utils.js

```diff
@@ -198,36 +194,33 @@ hqDefine("cloudcare/js/formplayer/utils/utils", function () {
         }
     };
 
-    Utils.getQueryInputs = function () {
-        return JSON.parse(sessionStorage.queryInputs || "{}");
+    Utils.getCurrentQueryInputs = function (queryKey) {
+        queryKey = queryKey || sessionStorage.queryKey;
+        const queryInputs = this.queryInputs || {};
+        return queryInputs[queryKey];
     };
 
-    Utils.setQueryInputs = function (inputs) {
-        const queryInputs = Utils.getQueryInputs();
-        queryInputs[sessionStorage.queryKey] = inputs;
-        sessionStorage.queryInputs = JSON.stringify(queryInputs);
-    };
-
-    Utils.getCurrentQueryInputs = function () {
-        const queryInputs = Utils.getQueryInputs();
-        return queryInputs[sessionStorage.queryKey] || {};
+    Utils.setCurrentQueryInputs = function (inputs, queryKey) {
+        queryKey = queryKey || sessionStorage.queryKey;
+        if (queryKey && queryKey !== "null" && queryKey !== "undefined") {
```

Comment:
Bit of discussion with @Jtang-1 on this above: https://github.com/dimagi/commcare-hq/pull/33946#discussion_r1449366455

Basically, without changing the logic of how/when `sessionStorage.queryKey` gets set which seems outside the scope of this PR, it sometimes gets set from a JS `undefined` or `null` to `"undefined"` or `"null"` since sessionStorage only holds strings. I could remove the check, it's not technically necessary, but it seemed nicer to avoid storing values we can't accurately look up later.